src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/MiscFilesProject.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.ProjectSystem;
+using Microsoft.AspNetCore.Razor.Utilities;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
+
+internal static class MiscFilesProject
+{
+    public static HostProject HostProject { get; }
+    public static string DirectoryPath { get; }
+
+    public static ProjectKey Key => HostProject.Key;
+    public static string FilePath => HostProject.FilePath;
+
+    static MiscFilesProject()
+    {
+        var tempDirectory = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString("n"));
```

Comment:
💡 We don't need a valid temp path do we?

Reply:
Nope, and we never have. This code was refactored from here: https://github.com/dotnet/razor/blob/ed0bec16620f3a3c17436f7fb2c572e02353c082/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/MiscFilesHostProject.cs#L37-L50

Reply:
In logs etc it probably would be easier if the path was shorter and not an actual path. Definitely doesn't have to be in this change, and I can do it later. It would make looking through logs in VS Code slightly more pleasant

Reply:
I think there are just a couple of requirements:

1. It probably needs to at least _look_ like a real path so that file normalization and `System.IO.Path` functions work correctly on it.
2. It should be unique on the system.

This code has been present since before my time on Razor, but it can certainly be improved.

Reply:
Filed #11296 to track improving this path.